### DISCUSSION TITLE
Responsive image Performance problem

### DISCUSSION THREAD

**Discussion by ariaieboy** [spatie/laravel-medialibrary#3306] [Ideas]
The problem is that this function [getSrcset](https://github.com/spatie/laravel-medialibrary/blob/7bde2a19cabfe808ab2a1b5223297f9988ecc42f/src/ResponsiveImages/RegisteredResponsiveImages.php#LL33C24-L33C24) adds around 10ms for each image that's going to render in the page.

I removed this function from `responsiveImage.blade.php` and I calculated these numbers.

10ms might be a small number, but when you have many images on one page, the response time increases quickly.

**Comment by freekmurze** [comment]
Feel free to create a PR that improves performance around this.

**Comment by ariaieboy** [comment]
I worked on this issue a while ago and figured out that the problem is with the S3 URL generator. It sends a request to the S3 server to get the File URL. 
For my project, I created a URL generator that generates file URLs without sending any request to the S3 server, but that will only work in some environments.
Adding a cache layer may work, but I need a better solution to make a PR for it.
